client/src/components/Users/Auth/SignUpCitizen/StepOne.tsx

```diff
@@ -0,0 +1,268 @@
+    rule: boolean().required('Required.'),
+  },
+};
+
+const getError = (rule: StringSchema | BooleanSchema, value: string | undefined) => {
+  try {
+    rule.validateSync(value);
+    return null;
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return error.message;
+    }
+  }
+};
+
+interface StepOneType {
+  initData: {};
+  handleNext: (formData: {}) => void;
+}
+
+export default function StepOne({ initData, handleNext }: StepOneType) {
+  const { classes } = useStyles();
+
+  Object.keys(initialFormData).forEach((key) => {
+    //@ts-ignore
+    initialFormData[key].value = initData[key];
+  });
+
+  const [formData, setFormData] = useState(initialFormData);
+
+  const nextEnabled = Object.values(formData).every(
+    ({ value, error }) => error === null && Boolean(value),
+  );
+
+  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value, checked }: { name: string; value: string; checked: boolean } = evt.target;
+    setFormData((fData) => {
+      return {
+        ...fData,
+        [name]: {
+          // @ts-ignore
+          ...fData[name],
+          value: name === 'acceptTerms' ? checked : value,
+        },
+      };
+    });
+  };
+
+  const handleBlur = (evt: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name }: { name: string } = evt.target;
+    if (Object.keys(initialFormData).includes(name)) {
+      setFormData((currFormData) => {
+        let error = null;
+        if (name === 'passwordConfirm') {
+          const passwordConfirmError = getError(
+            currFormData.passwordConfirm.rule,
+            currFormData.passwordConfirm.value,
+          );
+          error = passwordConfirmError;
+          if (currFormData.password.value !== currFormData.passwordConfirm.value) {
+            error = `Password mismatch.`;
+          }
+        } else {
+          // @ts-ignore
+          error = getError(currFormData[name].rule, currFormData[name].value);
+        }
+        return {
+          ...currFormData,
+          [name]: {
+            // @ts-ignore
+            ...currFormData[name],
+            error,
+          },
+        };
+      });
+    }
+  };
+
+  const handleClickNext = () => {
+    const formValues = Object.entries(formData).reduce((acc, [key, val]) => {
+      // @ts-ignore
+      acc[key] = val.value;
+      return acc;
+    }, {});
+    handleNext(formValues);
+  };
+
+  return (
+    <>
+      <Box sx={{ height: '100%', width: '100%' }}>
+        <Typography
+          sx={{ color: '#674E67' }}
+          className={classes.header}
+          lineHeight="87px"
+          fontSize="58px"
+          component="h1"
+          align="left"
+          variant="h4"
+          gutterBottom
+        >
+          Let's get started
+        </Typography>
+        <Typography
+          sx={{ color: '#323232' }}
+          className={classes.subHeader}
+          lineHeight="1.4"
+          fontSize="22px"
```

Comment:
I think it might be better to set these in `typography.ts`, maybe set a form input variant.